test/apps/iodemo/io_demo.cc

```diff
@@ -935,15 +927,22 @@ class DemoClient : public P2pDemoCommon {
         for (size_t server_index = 0; server_index < _server_info.size();
              ++server_index) {
             server_info_t& server_info = _server_info[server_index];
-            if ((server_info.conn != NULL) ||
-                (server_info.retry_count >= opts().client_retries)) {
-                /* server is already connected, or retry count exceeded */
+            if (server_info.conn != NULL) {
+                // Server is already connected
                 continue;
             }
 
+            // If retry count exceeded for at least one server, we should have
+            // exited already
+            assert(_status == OK);
+            assert(server_info.retry_count < opts().client_retries);
+
             server_info.conn = connect(opts().servers[server_index]);
             if (server_info.conn == NULL) {
                 connect_failed(server_index);
+                if (!_status != OK) {
+                    break;
```

Comment:
1. yep..
2. if one server exceeded we want to exit - since no point to continue running the test if one of the servers is not going to be used any more. if we could not reconnect to some server after N times it should be fatal

Reply:
2. I agree this situation. But as I mentioned in another comment, this is what we wanted to change originally. As I remember Alina reported the issue that client couldn't keep running with alive servers when some servers are killed. Also see the title of this pull request.

Reply:
the problem was when one server is killed (and restored) we closed all connections
IMO we can always keep the retry count as default (infinite) to keep retrying